File: subsystems/IntakeSubsystem.java
package org.firstinspires.ftc.teamcode.subsystems;

import java.util.function.Supplier;

import dev.nextftc.core.commands.Command;
import dev.nextftc.core.commands.utility.InstantCommand;
import dev.nextftc.core.commands.utility.LambdaCommand;
import dev.nextftc.core.subsystems.Subsystem;
import dev.nextftc.hardware.impl.MotorEx;
import dev.nextftc.hardware.powerable.SetPower;

public class IntakeSubsystem implements Subsystem {
    public static final IntakeSubsystem INSTANCE = new IntakeSubsystem();
    public MotorEx intakeMotor = new MotorEx("intakeMotor");
    private IntakeSubsystem() { }

    public Command setPower(Supplier<Double> leftTrigger, Supplier<Double> rightTrigger) {
        return new InstantCommand(() ->
                intakeMotor.setPower(rightTrigger.get() - leftTrigger.get()));
    }

    public Command setPower(double power) {
        return new InstantCommand(() ->
                intakeMotor.setPower(power));
    }

    public Command in = new SetPower(intakeMotor, 1);// power val is placeholder

    public Command out = new SetPower(intakeMotor, -1);// power val is placeholder
    public Command stop = new SetPower(intakeMotor, 1);// power val is placeholder
}

File: subsystems/lower/IntakeSubsystem.kt
package org.firstinspires.ftc.teamcode.subsystems.lower

import dev.nextftc.core.commands.Command
import dev.nextftc.core.subsystems.Subsystem
import dev.nextftc.hardware.impl.MotorEx
import dev.nextftc.hardware.powerable.SetPower
import java.util.function.Supplier

object IntakeSubsystem : Subsystem {
    private val motor = MotorEx("intake");

    val REVERSE = -1.0;
    val FORWARD = 1.0;

    var power = 1.0
        set(value) {
            field = value.coerceIn(-1.0, 1.0);
        };

    override fun periodic() {
        SetPower(motor, power);
    }

    class DriverCommandDefaultOn(  // could be bad for power draw?
        private val reversePower: Supplier<Double>,
    ) : Command() {
        override val isDone = false;

        override fun update() {
            power = 1.0 - 2.0 * reversePower.get();
        }
    }

    class DriverCommand(
        private val forwardPower: Supplier<Double>,
        private val reversePower: Supplier<Double>,
    ) : Command() {
        override val isDone = false;

        override fun update() {
            power = forwardPower.get() - reversePower.get();
        }
    }
}